.github/workflows/security.yml

```diff
@@ -17,30 +17,43 @@ defaults:
     shell: bash
 
 jobs:
+  update-vulnerability-database:
+    name: Trivy - Update cached database
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Download Trivy DB
+        id: db_download
+        run: trivy fs --download-db-only --cache-dir /home/runner/vuln-cache
+        continue-on-error: true
+
+      - name: Cache Trivy vulnerability database
+        if: ${{ steps.db_download.outcome == 'success' }}
+        uses: actions/cache@3624ceb22c1c5a301c8db4169662070a689d9ea8 # v4.1.1
+        with:
+          path: /home/runner/vuln-cache
+          key: trivy-cache-${{ github.run_id }}
+          restore-keys: |
+            trivu-latest-cache
+
   trivy-repo:
     name: Trivy - Repository
     runs-on: ubuntu-22.04
+    needs: update-vulnerability-database
```

Comment:
Can we do the same as in https://github.com/canonical/microcloud/pull/485 using the `restore-keys:` option?

Reply:
Yes, the current implementation is actually a product of my misunderstanding of the `restore-keys` option.
The changes are pushed and now it works similarly to MicroCloud (the correct way to use it).
I will test this first and then ping you for another review.